cookbooks/aws-parallelcluster-slurm/recipes/config_slurm_accounting.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-slurm
+# Recipe:: config_head_node
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+file "#{node['cluster']['slurm']['install_dir']}/etc/slurm_parallelcluster_slurmdbd.conf" do
+  owner "#{node['cluster']['slurm']['dbduser']}"
+  group "#{node['cluster']['slurm']['dbdgroup']}"
+  mode '0600'
+end
+
+template "#{node['cluster']['scripts_dir']}/slurm/update_slurm_database_password.sh" do
+  source 'slurm/head_node/update_slurm_database_password.sh.erb'
+  owner 'root'
+  group 'root'
+  mode '0700'
+  variables(
+    secret_arn: node['cluster']['slurm']['database']['password_secret_arn'],
+    region: node['cluster']['region']
+  )
+  sensitive true
+end
+
+execute "update Slurm database password" do
+  user 'root'
+  group 'root'
+  command "#{node['cluster']['scripts_dir']}/slurm/update_slurm_database_password.sh"
+end
+
+service "slurmdbd" do
+  supports restart: false
+  action %i(enable start)
+end
+
+# After starting slurmdbd the database may not be fully responsive yet and
+# its bootstrapping may fail. We need to wait for sacctmgr to successfully
+# query the database before proceeding.
+bash "wait for slurm database" do
```

Comment:
1. are there cases when this can take much longer than 30seconds? e.g. using a db that had been already used with an older Slurm version? is this something we need to account for?
2. you can look at how a similar check is don in head_node_finalize, the useful part about that approach is that it leaves a trace in the chef logs about what it's doing so that it's easier to debug issues

Reply:
I had originally put a sleep of 10 seconds, and sometimes it worked, sometimes it didn't. So I put a more robust wait loop.

> e.g. using a db that had been already used with an older Slurm version? is this something we need to account for? 

Good catch. Indeed from the [Slurm documentation](https://slurm.schedmd.com/quickstart_admin.html#upgrade):

> Database table changes may be required for the upgrade, for example adding new fields to existing tables. If the database contains a large number of entries, the SlurmDBD daemon may require tens of minutes to update the database and be unresponsive during this time interval.

Should we consider this scenario here? For such an operation we should probably recommend to use a non-production `slurmdbd` to update the database (similar to the scenario of having a backup of old data in a non-production `slurmdbd`). What do you think? If you want I can increase the total wait time to something like 5 minutes (30 retries with 10 sleep time)?

> you can look at how a similar check is don in head_node_finalize
Yes, that's not bad. I'll implement it this way.

Reply:
Reimplemented the wait with a Chef `execute` resource.

Reply:
Fdm and I decided to remove the timeout for the time being and rely on the CFN default timeout.

Reply:
The most straightforward way to implement this with the `execute` Chef resource is to set a very large number for the `retries parameter. Setting 100000 retries of 10 seconds each.